Vienna/Sources/Main window/BrowserTab+Interface.swift

```diff
@@ -140,13 +140,13 @@ extension BrowserTab {
     }
 
     func updateTabTitle() {
-        if self.url != webView.url || self.url == nil {
+        if self.url != webView.url || webView.url == nil {
             // Currently loading (the first time), webview title not yet correct / available
             self.title = self.url?.host ?? NSLocalizedString("New Tab", comment: "")
         } else if let title = self.webView.title, !title.isEmpty {
             self.title = title
-        } else {
-            // Webview is about:blank or empty
+        } else if let empty = self.title?.isEmpty, empty {
```

Comment:
I suspect if the title is nil, the case should also be entered? Then you would need to write `if self.title == nil || self.title!.isEmpty`. Otherwise, I think it is more elegant to do `if let` on the component that can actually be null, i.e. the title.

Reply:
Please see updated code